mod/cli/pkg/commands/genesis/payload.go

```diff
@@ -87,17 +87,17 @@ func AddExecutionPayloadCmd() *cobra.Command {
 			}
 
 			genesisInfo := &genesis.Genesis[
-				*types.Deposit,
-				*types.ExecutionPayloadHeaderDeneb,
+				*types.Deposit, *types.ExecutionPayloadHeader,
 			]{}
+
 			if err = json.Unmarshal(
 				appGenesisState["beacon"], genesisInfo,
 			); err != nil {
 				return errors.Wrap(err, "failed to unmarshal beacon state")
 			}
 
 			// Inject the execution payload.
-			genesisInfo.ExecutionPayloadHeader, err =
+			denebHeader, err :=
 				executableDataToExecutionPayloadHeader(payload)
```

Comment:
should probably refactor this function to take a forkVersion and produce the real Header, can read the forkversion from genesis.Info